vars/runPerfTestScript.groovy

```diff
@@ -4,24 +4,28 @@ void call(Map args = [:]) {
 
     install_dependencies()
     install_opensearch_infra_dependencies()
-    config_name = isNullOrEmpty(args.conig) ? "config.yml" : args.config
+    config_name = isNullOrEmpty(args.config) ? "config.yml" : args.config
     withAWS(role: 'opensearch-test', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
         s3Download(file: "config.yml", bucket: "${ARTIFACT_BUCKET_NAME}", path: "${PERF_TEST_CONFIG_LOCATION}/${config_name}", force: true)
     }
 
-    sh([
-        './test.sh',
-        'perf-test',
-        args.insecure ? "--stack test-single-${args.buildId}-${args.architecture}" :
-        "--stack test-single-security-${args.buildId}-${args.architecture}",
-        "--bundle-manifest ${args.bundleManifest}",
-        "--config config.yml",
-        args.insecure ? "--without-security" : "",
-        isNullOrEmpty(args.workload) ? "" : "--workload ${args.workload}",
-        isNullOrEmpty(args.testIterations) ? "" : "--test-iters ${args.testIterations}",
-        isNullOrEmpty(args.warmupIterations) ? "" : "--warmup-iters ${args.warmupIterations}",
-        isNullOrEmpty(args.component) ? "" : "--component ${args.component}"
-    ].join(' '))
+    String stackNameSuffix = isNullOrEmpty(args.stackNameSuffix) ? 'perf-test' : args.stackNameSuffix
+
+    withCredentials([usernamePassword(credentialsId: "${GITHUB_BOT_TOKEN_NAME}", usernameVariable: 'GITHUB_USER', passwordVariable: 'GITHUB_TOKEN')]) {
```

Comment:
I dont see anywhere define the actual token id, are you setting it in the jenkins global env?

Reply:
It is defined as a Jenkins global env var. More usages - https://github.com/opensearch-project/opensearch-build/blob/c125b697815ca9887873752e0c3c7bdda0dfe00d/vars/createReleaseTag.groovy#L12, https://github.com/opensearch-project/opensearch-build/blob/de56d0fef90962339e4b292890c69c5948a3663a/vars/signArtifacts.groovy#L28